src/Routes/routes_fastapi.py

```
@@ -402,10 +409,14 @@ def save_route(
-        return {
-            "success" : False,
-            "message" : "A route with this name already exists. Please pick a new name."
-        }
+        raise HTTPException(
+            status_code=422,
+            detail={
+                "success": False,
+                "message": "ERROR : route name cannot be the same as other routes.",
+                "user_message": "A route with this name already exists. Please pick a new name."
+            }
+        )
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)